test_conformance/extensions/cl_khr_command_buffer/command_buffer_get_command_buffer_info.cpp

```diff
@@ -0,0 +1,339 @@
+                &state, nullptr);
+            test_error_ret(error, "clGetCommandBufferInfoKHR failed",
+                           TEST_FAIL);
+
+            test_expected_info(state != expected);
+
+            return TEST_PASS;
+        };
+
+        // verify recording state
+        error = verify_state(CL_COMMAND_BUFFER_STATE_RECORDING_KHR);
+        test_error(error, "verify_state failed");
+
+        // record command buffer
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        // verify executable state
+        error = verify_state(CL_COMMAND_BUFFER_STATE_EXECUTABLE_KHR);
+        test_error(error, "verify_state failed");
+
+        error = clEnqueueFillBuffer(queue, out_mem, &pattern, sizeof(cl_int), 0,
+                                    data_size(), 0, nullptr, nullptr);
+        test_error(error, "clEnqueueFillBuffer failed");
+
+        clEventWrapper trigger_event = clCreateUserEvent(context, &error);
+        test_error(error, "clCreateUserEvent failed");
+
+        // enqueued command buffer blocked on user event
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 1,
+                                          &trigger_event, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        // verify pending state
+        error = verify_state(CL_COMMAND_BUFFER_STATE_PENDING_KHR);
+        test_error(error, "verify_state failed");
+
+        // execute command buffer
+        error = clSetUserEventStatus(trigger_event, CL_COMPLETE);
+        test_error(error, "clSetUserEventStatus failed");
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RunPropArrayInfoTest()
+    {
+        cl_int error = CL_SUCCESS;
+
+        // record command buffer
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        size_t ret_value_size = 0;
+        std::vector<cl_command_buffer_properties_khr> combuf_props;
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR, 0, nullptr,
+            &ret_value_size);
+        test_error_ret(error, "clGetCommandBufferInfoKHR failed", TEST_FAIL);
+
+        test_expected_info(ret_value_size == 0);
+
+        cl_uint num_ret_props =
+            ret_value_size / sizeof(cl_command_buffer_properties_khr);
+        combuf_props.resize(num_ret_props);
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR,
+            num_ret_props * sizeof(cl_command_buffer_properties_khr),
+            combuf_props.data(), nullptr);
+        test_error_ret(error, "clGetCommandBufferInfoKHR failed", TEST_FAIL);
+
+        for (int i = 0; i < num_ret_props; i++)
+            if (combuf_props[i] == CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR)
```

Comment:
understood, then we must memorize properties array for default command_buffer in `BasicCommandBufferTest` structure, right ?

We could also make a bald assumption that below code from base structure `BasicCommandBufferTest` will stay intact. In that case we could even skip iteration and simply check if there is only one property and it is equal `CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR`

```
    if (simultaneous_use_support)
    {
        cl_command_buffer_properties_khr properties[3] = {
            CL_COMMAND_BUFFER_FLAGS_KHR, CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR,
            0
        };
        command_buffer =
            clCreateCommandBufferKHR(1, &queue, properties, &error);
    }

```

Reply:
How things are currently setup in the command-buffer tests, if we want to test a command-buffer with more specific properties, then the derived test fixture should set that up themselves (e.g out-of-order command-buffer tests). And `BasicCommandBufferTest` will create these basic command-buffer for more generic tests. In the future it may be that we are smarter about setting properties in `BasicCommandBufferTest` so that some device specific flags are added to the passed properties based on query results. However, that's not how the command-buffer tests are currently setup, and should be a PR addressing that concern in isolation since it will likely affect other tests too. Therefore, for the scope of this PR I think it's fine to assume that the properties passed in `BasicCommandBufferTest`
will stay intact. 

So we could indeed skip iterating during validation as you describe. We should also check another case however (the `else` to the line i've commented on), that if simultaneous use  wasn't set, then the correct value is returned to stress this spec wording

> If the properties argument specified in clCreateCommandBufferKHR used to
create command_buffer was NULL, the implementation may return either a param_value_size_ret
of 0 (i.e. there is are no properties to be returned), or the implementation may return
a property value of 0 (where 0 is used to terminate the properties list).

Reply:
> ... We should also check another case however (the else to the line i've commented on), that if simultaneous use wasn't set, then the correct value is returned to stress this spec wording

This changes the logic of test a bit. So far `Skip` method verified if `CombufInfoTestMode::CITM_PROP_ARRAY` test created command buffer with simultaneous use, otherwise test skipped. After correction this test accepts both scenarios (`NULL` and simultaneous use prop array).